core/src/main/java/com/zextras/carbonio/user_management/services/UserService.java

```diff
@@ -136,6 +142,53 @@ public Response getInfoByEmail(
     return Response.ok().entity(userInfo).build();
   }
 
+  public Optional<UserMyself> getMyselfByToken(String token) {
+    try {
+      GetInfoResponse infoResponse = SoapClient
+        .newClient()
+        .setAuthToken(token)
+        .getAccountInfoByAuthToken();
+
+      UserId userId = new UserId();
+      userId.setUserId(infoResponse.getId());
+
+      String locale = infoResponse
+        .getPrefs()
+        .getPref()
+        .stream()
+        .filter(perf -> perf.getName().equals("zimbraPrefLocale"))
+        .findFirst()
+        .map(Pref::getValue)
+        .orElse("en");
+
+      String fullName = infoResponse
+        .getAttrs()
+        .getAttr()
+        .stream()
+        .filter(attribute -> attribute.getName().equals("displayName"))
+        .findFirst()
+        .map(Attr::getValue)
+        .orElse("");
+
+      UserMyself userMyself = new UserMyself();
+      userMyself.setId(userId);
+      userMyself.setEmail(infoResponse.getName());
+      userMyself.setDomain(infoResponse.getPublicURL());
+      userMyself.setFullName(fullName);
+      userMyself.setLocale(Locale.valueOf(locale.toUpperCase()));
+
+      return Optional.of(userMyself);
+
+    } catch (ServerSOAPFaultException exception) {
+      System.out.println(exception.getMessage());
```

Comment:
We setup the logback module, shouldn't we log this instead of printing?

Reply:
We are doing it with on a separate task where we will replace every `System.out` with a logger